File: StudentOglasi/StudentOglasi.Services/Database/SmjestajnaJedinica.cs
using System;
using System.Collections.Generic;

namespace StudentOglasi.Services.Database;

public partial class SmjestajnaJedinica
{
    public int Id { get; set; }

    public string Naziv { get; set; } = null!;

    public decimal Cijena { get; set; }

    public int Kapacitet { get; set; }

    public string? Opis { get; set; }

    public string? DodatneUsluge { get; set; }

    public bool? Kuhinja { get; set; }

    public bool? Tv { get; set; }

    public bool? KlimaUredjaj { get; set; }

    public bool? Terasa { get; set; }

    public int? SmjestajId { get; set; }

    public virtual ICollection<Slike> Slikes { get; set; } = new List<Slike>();

    public virtual Smjestaji? Smjestaj { get; set; }
}

File: StudentOglasi/UI/studentoglasi_admin/lib/models/SmjestajnaJedinica/smjestajna_jedinica.dart
import 'package:file_picker/file_picker.dart';
import 'package:json_annotation/json_annotation.dart';
import 'package:studentoglasi_admin/models/Slike/slike.dart';

part 'smjestajna_jedinica.g.dart';

@JsonSerializable()
class SmjestajnaJedinica {
  int? id;
  String? naziv;
  double? cijena;
  int? kapacitet;
  String? opis;
  bool? kuhinja;
  bool? tv;
  bool? klimaUredjaj;
  bool? terasa;
  String? dodatneUsluge;
  int? smjestajId;
  List<String>? slike;
  List<Slike>? slikes;

  SmjestajnaJedinica(
      this.id,
      this.naziv,
      this.cijena,
      this.kapacitet,
      this.opis,
      this.kuhinja,
      this.tv,
      this.klimaUredjaj,
      this.terasa,
      this.dodatneUsluge,
      this.smjestajId,
      this.slike,
      this.slikes);

  factory SmjestajnaJedinica.fromJson(Map<String, dynamic> json) =>
      _$SmjestajnaJedinicaFromJson(json);

  Map<String, dynamic> toJson() => _$SmjestajnaJedinicaToJson(this);
}
